### PR TITLE
feat: add driver update cli command

### PR DESCRIPTION
**Pull request by jlipps**

### CODE REVIEW

File: lib/cli/args.js
Comment (mykola-mokhnach) [inline lib/cli/args.js:390]: major and beta a different things. I would rather put them as separate options

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:85]: we could combine this condition with the previous one using ||

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:278]: how about giving a more descriptive name to this var (`shouldUpdateAll`)

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:294]: const {safeUpdate, unsafeUpdate} = ...

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:306]: do we need to enclose the lambda body into curly brackets here?

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:284]: it makes sense to either add a comment about what will be keys ands values of these maps

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:322]: please move magic strings to constants. This might break easily if one changes the original message

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:338]: it would be nice to have some unit tests for this logic

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:354]: docstrings would be nice

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:279]: We usually use the [typedef](https://github.com/appium/appium-adb/blob/master/lib/tools/adb-commands.js#L965) format to describe such things

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:284]: `{Object}`

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:308]: does it make sense to have different error classes instead?

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:340]: if ([NO_UPDATES_MSG, NOT_NPM_MSG].map((x) -> new RegExp(_.escapeRegExp(x)).some((p) -> p.test(err)))

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:356]: Please use curly brackets for types to align with the common documentation style

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:358]: ⬆️

File: test/cli-specs.js
Comment (mykola-mokhnach) [inline test/cli-specs.js:1]: 😍

File: lib/cli/driver.js
Comment (mykola-mokhnach) [inline lib/cli/driver.js:12]: should we have documented somewhere the set of rules for drivers naming and which of these names are reserved?

File: lib/cli/npm.js
Comment (mykola-mokhnach) [inline lib/cli/npm.js:71]: Please add `return` description

Comment (mykola-mokhnach) [inline lib/cli/npm.js:71]: `@param {Array<string>} allVersions`

**Review by mykola-mokhnach — APPROVED** [review]
Had some minor comments, but in general I like it

**Review by KazuCocoa — APPROVED** [review]

File: lib/cli/driver.js
Comment (jlipps) [reply lib/cli/driver.js:12]: yeah, this should be in the documentation. i don't see this as dangerous though, if someone creates a driver called 'installed' it will just never be updatable using this command. would be a dumb move though.

also i don' tmind if people create drivers with the same name, just whichever one is installed first will be the one that's used, and others won't be able to be installed.

**Comment by jlipps** [comment]
ok, all feedback addressed. i think this is good to go? once it's merged, i'll put together a short driver cli doc, and then we can release a 2.0 beta to start seeing where the issues might be with this approach.

**Comment by jlipps** [comment]
any more thoughts @mykola-mokhnach or is this good to go?

**Pull request merged**
